File: postgresql-async/src/main/scala/com/github/mauricio/async/db/postgresql/parsers/BackendKeyDataParser.scala
/*
 * Copyright 2013 Maurício Linhares
 *
 * Maurício Linhares licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

package com.github.mauricio.async.db.postgresql.parsers

import com.github.mauricio.async.db.postgresql.messages.backend.{ProcessData, ServerMessage}
import io.netty.buffer.ByteBuf

object BackendKeyDataParser extends MessageParser {

  override def parseMessage(b: ByteBuf): ServerMessage = {
    new ProcessData(b.readInt(), b.readInt())
  }

}

File: postgresql-async/src/main/java/com/github/mauricio/async/db/postgresql/parsers/BackendKeyDataParser.kt
package com.github.mauricio.async.db.postgresql.parsers

import com.github.mauricio.async.db.postgresql.messages.backend.ProcessData
import com.github.mauricio.async.db.postgresql.messages.backend.ServerMessage
import io.netty.buffer.ByteBuf


object BackendKeyDataParser : MessageParser {

  override fun parseMessage(b: ByteBuf): ServerMessage {
    return ProcessData(b.readInt(), b.readInt())
  }

}
